tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/ExtraEntrySortedMap.java

```diff
@@ -0,0 +1,207 @@
+        if (ordering.compare(fromKey, extraKey) <= 0 && ordering.compare(toKey, extraKey) > 0) {
+            return new ExtraEntrySortedMap<>(newBase, extraKey, extraValue);
+        }
+        return newBase;
+    }
+
+    @Override
+    public SortedMap<K, V> headMap(K toKey) {
+        SortedMap<K, V> newBase = base.headMap(toKey);
+        if (ordering.compare(toKey, extraKey) > 0) {
+            return new ExtraEntrySortedMap<>(newBase, extraKey, extraValue);
+        }
+        return newBase;
+    }
+
+    @Override
+    public SortedMap<K, V> tailMap(K fromKey) {
+        SortedMap<K, V> newBase = base.tailMap(fromKey);
+        if (ordering.compare(fromKey, extraKey) <= 0) {
+            return new ExtraEntrySortedMap<>(newBase, extraKey, extraValue);
+        }
+        return newBase;
+    }
+
+    @Override
+    public K firstKey() {
+        if (base.isEmpty()) {
+            return extraKey;
+        }
+        return ordering.min(base.firstKey(), extraKey);
+    }
+
+    @Override
+    public K lastKey() {
+        if (base.isEmpty()) {
+            return extraKey;
+        }
+        return ordering.max(base.lastKey(), extraKey);
+    }
+
+    @Override
+    public int size() {
+        return base.size() + 1;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return false;
+    }
+
+    @Override
+    public boolean containsKey(Object key) {
+        return extraKey.equals(key) || base.containsKey(key);
+    }
+
+    @Override
+    public boolean containsValue(Object value) {
+        return extraValue.equals(value) || base.containsValue(value);
+    }
+
+    @Override
+    public V get(Object key) {
+        if (extraKey.equals(key)) {
+            return extraValue;
+        }
+        return base.get(key);
+    }
+
+    @Override
+    public V put(K key, V value) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public V remove(Object key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void putAll(Map<? extends K, ? extends V> other) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void clear() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Set<K> keySet() {
+        return new AbstractSet<K>() {
+            @Override
+            public Iterator<K> iterator() {
+                return Iterables.mergeSorted(ImmutableList.of(base.keySet(), ImmutableList.of(extraKey)), ordering)
+                        .iterator();
+            }
+
+            @Override
+            public int size() {
+                return base.size() + 1;
+            }
+        };
+    }
+
+    @Override
+    public Collection<V> values() {
+        return new AbstractCollection<V>() {
+            @Override
+            public Iterator<V> iterator() {
+                return keySet().stream().map(ExtraEntrySortedMap.this::get).iterator();
```

Comment:
can avoid compiler warning with:

```suggestion
                return Iterators.transform(keySet().iterator(), key -> get(key));
```